added support for specifying json convenience object

blagerweij opened the pull request:
Allows you to conveniently use `json` field for multipart forms:

```
        http:
          url: https://httpbin.org/post
          method: POST
          formData:
            myfield:
              json:
                foo: bar
                bar: foo
```
The YAML will be converted to JSON, and the content-type set to 'application/json'.

mishushakov commented on src/steps/http.ts:
```diff
@@ -286,10 +291,6 @@ export default async function (
       const appendOptions = {} as FormData.AppendOptions
       if (typeof params.formData[field] === 'string') {
```
One thing: Can we also check for boolean and number? They should be converted automatically to string if you pass them to formData.append

blagerweij replied:
done, please check

mishushakov reviewed (commented):
Overall looks good to me and I'd be happy to merge. Feel free to act on my comment.

mishushakov commented:
Okay looks good to me now. Have you tested it?

blagerweij commented:
Yes, I have also added a 'jsonfield' in the mulltipart.yaml test-file.

mishushakov commented:
Do you actually use `HTTPRequestPartJson` type anywhere?

blagerweij commented:
@mishushakov nice catch! I had initially added that type, but later decided to combine the json field in the existing HttpRequestPart type. I've updated the PR (removed the unused code); Also added a testcase for JSON arrays. I've tested it locally, and for me it works.

mishushakov commented:
Very well done. One thing though: you can let the `!= null` part out, I don't think it makes any difference.

blagerweij commented:
What happens if you use the following (valid) json:

```
          my_json_field:
              json: false
```

If I'm not wrong, then `if (requestPart.json != null)` works as expected, where `if (requestPart.json)` does not ?!?

mishushakov commented:
Yes, the problem here will be if I want my json to be null. What you actually need is `if('json' in requestPart)`

mishushakov commented:
Thank you very much! We should also fix the same problem in other places.
